File: Python/A_Project/C_Elevator_System/ObserverPattern/elevator_display.py
from ObserverPattern.elevator_observer import ElevatorObserver


class ElevatorDisplay(ElevatorObserver):
    def on_elevator_state_change(self, elevator, state):
        state_name = getattr(state, "name", str(state))
        print(f"Elevator {elevator.get_elevator_id()} changed state to {state_name}")

    def on_elevator_floor_change(self, elevator, floor: int):
        print(f"Elevator {elevator.get_elevator_id()} arrived at floor {floor}")

File: Ruby/A_Project/C_Elevator_System/ObserverPattern/elevator_display.rb
require_relative 'elevator_observer'

# Concrete observer that prints elevator state and floor changes to stdout
class ElevatorDisplay < ElevatorObserver
  def on_elevator_state_change(elevator, state)
    puts "Elevator #{elevator.get_elevator_id} changed state to #{state}"
  end

  def on_elevator_floor_change(elevator, floor)
    puts "Elevator #{elevator.get_elevator_id} arrived at floor #{floor}"
  end
end
